src/go/k8s/pkg/resources/superusers.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package resources
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	redpandav1alpha1 "github.com/vectorizedio/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/api/admin"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const scramPandaproxyUsername = "pandaproxy_client"
+
+var _ Resource = &SuperUsersResource{}
+
+// SuperUsersResource is part of the reconciliation of redpanda.vectorized.io CRD
+// focusing on the super users for Schema Registry and Panda proxy
+type SuperUsersResource struct {
+	k8sclient.Client
+	scheme                  *runtime.Scheme
+	pandaCluster            *redpandav1alpha1.Cluster
+	fqdn                    string
+	setSuperUsersInRedpanda bool
+	logger                  logr.Logger
+}
+
+// NewSuperUsers creates SuperUsersResource that managed super users
+// for Schema Registry and Panda proxy
+func NewSuperUsers(
+	client k8sclient.Client,
+	pandaCluster *redpandav1alpha1.Cluster,
+	scheme *runtime.Scheme,
+	fqdn string,
+	setSuperUsersInRedpanda bool,
+	logger logr.Logger,
+) *SuperUsersResource {
+	return &SuperUsersResource{
+		client,
+		scheme,
+		pandaCluster,
+		fqdn,
+		setSuperUsersInRedpanda,
+		logger.WithValues(
+			"Kind", ingressKind(),
+		),
+	}
+}
+
+// Ensure will manage Super users for Panda proxy and Schema Registry for redpanda.vectorized.io custom resource
+func (r *SuperUsersResource) Ensure(ctx context.Context) error {
+	adminInternal := r.pandaCluster.AdminAPIInternal()
+	if adminInternal == nil {
+		return nil
+	}
+
+	adminInternalPort := adminInternal.Port
+
+	var urls []string
+	replicas := *r.pandaCluster.Spec.Replicas
+
+	for i := int32(0); i < replicas; i++ {
+		urls = append(urls, fmt.Sprintf("%s-%d.%s:%d", r.pandaCluster.Name, i, r.fqdn, adminInternalPort))
+	}
+
+	adminAPI, err := admin.NewAdminAPI(urls, nil)
+	if err != nil {
+		return fmt.Errorf("creating admin api: %w", err)
+	}
+
+	objs, err := r.objs()
+	if err != nil {
+		return fmt.Errorf("unable to construct object: %w", err)
+	}
+
+	var created bool
+	for _, obj := range objs {
+		if !r.setSuperUsersInRedpanda {
+			created, err = CreateIfNotExists(ctx, r, obj, r.logger)
+			if err != nil || created {
+				return err
+			}
+		} else {
+			var secret corev1.Secret
+			err = r.Get(ctx, types.NamespacedName{
```

Comment:
also I don't understand why in the first branch we create but here we expect that it already exists? 🤔 how do we know that?

Reply:
See https://github.com/vectorizedio/redpanda/pull/1893#discussion_r677422921